Fixed test_positive_local_directory_sync test cli test

omkarkhatavkar opened the pull request:
**Description** 
Fixed test_positive_local_directory_sync test cli test, this was failing due to unable to find the package python-plup-manifest on Satellite Host. Now Configuration is added such that this package will get downloaded and installed on Satellite host as part of test. 

**Test Result** 
```
[root@okhatavk robottelo]# pytest tests/foreman/cli/test_repository.py -v -k "test_positive_local_directory_sync"
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.4.8, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /usr/bin/python3.4
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/okhatavk/Satellite/robottelo, inifile:
plugins: wait-for-1.0.9, xdist-1.22.2, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collecting 86 items                                                                                                                                                                          2018-08-07 17:23:27 - conftest - DEBUG - BZ deselect is disabled in settings

collected 86 items / 85 deselected                                                                                                                                                           

tests/foreman/cli/test_repository.py::FileRepositoryTestCase::test_positive_local_directory_sync PASSED                                                                                [100%]

========================================================================== 1 passed, 85 deselected in 81.71 seconds ==========================================================================


```

ldjebran commented:
as understood you are fixing a test, and that package is located at custom sat-tools compose repository, on my sat installation that repository was enabled by the automation and was able to install that package (python-pulp-manifest) without problem
```
# yum -y install python-pulp-manifest

==============================================================================================================
 Package   Arch   Version         Repository                                                             Size
==============================================================================================================
Installing:
 python-pulp-manifest
           noarch 2.16.3-1.el7sat Sat6-XXXXX{repo info removed}_x86_64  54 k

Transaction Summary
==============================================================================================================
Install  1 Package

Total download size: 54 k
Installed size: 23 k

Installed:
  python-pulp-manifest.noarch 0:2.16.3-1.el7sat                                                               

# pulp-manifest --help
usage: pulp-manifest [-h] directory

positional arguments:
  directory   A path to the directory where the PULP_MANIFEST file should be
              created.

optional arguments:
  -h, --help  show this help message and exit

```  
@kbidarkar @rplevka perhaps this is an infra issue,  as we have to maintain this package file name every time it change version number for every build with this implementation, but with the old implementation we do not care about exact package name, what do you think?

rplevka commented:
@ldjebran @omkarkhatavkar  - this is not an infra issue, i just assume that you guys tested on your setups installed using satellite installer jenkins job - the standalone job enables the sat tools repo by default, however this is not necessary for the production.
@omkarkhatavkar @sghai - i would completely avoid generating a new content + pulp manifest in the test - we already have test that covers syncing of the file repos: `test_positive_synchronize_file_repo` - this one simply takes the pre-created file repo (including pulp manifest) from 
```
CUSTOM_FILE_REPO = (
    u'https://repos.fedorapeople.org/repos/pulp/pulp/fixtures/file/'
)
```
- I don't really know whether another test that would basically do the very same thing excepting taht it places the repo to the localhost has any meaning or added value to us - however, if you still insist of having it, i would use the `CUSTOM_FILE_REPO` url to download the pre-created content to the destination and won't bother with fabricating a new repo every time.
- also i don't consider installing additional packages to the testing setup a good practice (on the other hand, i respect having pulp-admin-client installed on our sat, so i'm not really consistent here).

sghai commented:
@omkarkhatavkar : I think best would be to pull the file from http url as @rplevka pointed above to local file system and then use that file with `file:///` this will avoid fetching pulp-manifest package and rebuilding the file.

omkarkhatavkar commented:
As discussed, added fix and removed all other code changes are not required.

**Test Result** 

```
root@okhatavk robottelo (fix_test_positive_local_directory_sync) $ pytest tests/foreman/cli/test_repository.py -v -k "test_positive_file_repo_local"
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.4.8, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /usr/bin/python3.4
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/okhatavk/Satellite/robottelo, inifile:
plugins: wait-for-1.0.9, xdist-1.22.2, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collecting 86 items                                                                                                                                                                          2018-08-09 20:47:23 - conftest - DEBUG - BZ deselect is disabled in settings

collected 86 items / 85 deselected                                                                                                                                                           

tests/foreman/cli/test_repository.py::FileRepositoryTestCase::test_positive_file_repo_local_directory_sync PASSED                                                                      [100%]

========================================================================== 1 passed, 85 deselected in 69.26 seconds ==========================================================================
```
